addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoItemValueConverter.java

```diff
@@ -0,0 +1,170 @@
+            if (StringUtils.isEmpty(format)) {
+                return new NeeoItemValue(((DecimalType) state).toBigDecimal());
+            }
+        } else if (state instanceof UnDefType) {
+            return new NeeoItemValue("-");
+        } else if (state instanceof StringType && channel.getType() == NeeoCapabilityType.IMAGEURL) {
+            return new NeeoItemValue(state.toString());
+        }
+
+        String itemValue;
+        if (StringUtils.isNotEmpty(format)) {
+            if (state == null || state instanceof UnDefType) {
+                itemValue = formatUndefined(format);
+            } else if (state instanceof Type) {
+                if (TransformationHelper.isTransform(format)) {
+                    try {
+                        final String transformed = TransformationHelper
+                                .transform(context.getComponentContext().getBundleContext(), format, state.toString());
+                        return new NeeoItemValue(transformed);
+                    } catch (NoClassDefFoundError ex) {
+                        // TransformationHelper is optional dependency, so ignore if class not found
+                    }
+                }
+
+                // The following exception handling has been added to work around a Java bug with formatting
+                // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
+                // Without this catch, the whole sitemap, or page can not be displayed!
+                // This also handles IllegalFormatConversionException, which is a subclass of IllegalArgument.
+                try {
+                    itemValue = ((Type) state).format(format);
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Exception while formatting value '{}' of item {} with format '{}': {}", state,
+                            channel.getItemName(), format, e);
+                    itemValue = new String("Err");
+                }
+            } else {
+                itemValue = state.toString();
+            }
+        } else {
+            itemValue = state.toString();
+        }
+
+        return new NeeoItemValue(itemValue);
+    }
+
+    /**
+     * Takes the given <code>formatPattern</code> and replaces it with a analog
+     * String-based pattern to replace all value Occurrences with a dash ("-").
+     *
+     * @param formatPattern the original pattern which will be replaces by a
+     *            String pattern.
+     * @return a formatted String with dashes ("-") as value replacement
+     */
+    private static String formatUndefined(String formatPattern) {
+        String undefinedFormatPattern = formatPattern.replaceAll(IDENTIFY_FORMAT_PATTERN_PATTERN, "%1\\$s");
+        try {
+            return String.format(undefinedFormatPattern, "-");
+        } catch (Exception e) {
+            final Logger logger = LoggerFactory.getLogger(NeeoItemValueConverter.class);
```

Comment:
Yeah - I was trying to get around the maven warning about no static loggers.  Of course, since the class has an instance logger and this method is static - you then get the 'multiple logger' warning.  Personally - I'd go for static logger in this case (which I'll change pending @kaikreuzer )